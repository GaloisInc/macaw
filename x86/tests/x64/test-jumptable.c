
int a(int x);
int b();
int c();
int d();
int e();

// A 0-based jump table
int switch_zero_based(int i) {
    switch (i) {
    case 0:
	return a(1);
    case 1:
	return b();
    case 2:
	return c();
    case 3:
	return d();
    case 4:
	return e();
    case 5:
	return 5;
    case 6:
	return 1123213;
    case 7:
	return 191286;
    case 8:
	return 921312;
    default:
	return 0;
    }
}

// A one-based jump table
int switch_one_based(int i) {
    switch (i) {
    case 1:
	return b();
    case 2:
	return c();
    case 3:
	return d();
    case 4:
	return e();
    case 5:
	return 5;
    case 6:
	return 1123213;
    default:
	return 0;
    }
}
