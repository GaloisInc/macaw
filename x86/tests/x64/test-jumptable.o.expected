R { funcs = [(Addr 1 0
    	     , [(Addr 1 0x0,37)
               ,(Addr 1 0x25,14)
               ,(Addr 1 0x33,10)
               ,(Addr 1 0x3d,8)
               ,(Addr 1 0x45,7)
               ,(Addr 1 0x4c,8)
               ,(Addr 1 0x54,7)
               ,(Addr 1 0x5b,8)
               ,(Addr 1 0x63,7)
               ,(Addr 1 0x6a,8)
               ,(Addr 1 0x72,7)
               ,(Addr 1 0x79,8)
               ,(Addr 1 0x81,12)
               ,(Addr 1 0x8d,12)
               ,(Addr 1 0x99,12)
               ,(Addr 1 0xa5,12)
               ,(Addr 1 0xb1,7)
               ,(Addr 1 0xb8,9)])
            ,(Addr 1 0xd0
    	     , [(Addr 1 0xd0,35)
               ,(Addr 1 0xf3,14)
               ,(Addr 1 0x101,7)
               ,(Addr 1 0x108,8)
               ,(Addr 1 0x110,7)
               ,(Addr 1 0x117,8)
               ,(Addr 1 0x11f,7)
               ,(Addr 1 0x126,8)
               ,(Addr 1 0x12e,7)
               ,(Addr 1 0x135,8)
               ,(Addr 1 0x13d,12)
               ,(Addr 1 0x149,12)
               ,(Addr 1 0x155,7)
               ,(Addr 1 0x15c,9)])
            ]
  , ignoreBlocks = []
  }
